components/public-api-server/pkg/server/server.go

```diff
@@ -133,7 +133,7 @@ func Start(logger *logrus.Entry, version string, cfg *config.Configuration) erro
 	if redisClient == nil {
 		return fmt.Errorf("no Redis configiured")
 	}
-	idpService, err := identityprovider.NewService(strings.TrimSuffix(cfg.PublicURL, "/")+"/idp", identityprovider.NewRedisCache(redisClient))
+	idpService, err := identityprovider.NewService(strings.TrimSuffix(cfg.PublicURL, "/")+"/idp", identityprovider.NewRedisCache(ctx, redisClient))
```

Comment:
The ctx you're using here is setup with 
```
ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
	defer cancel()
	err = redisClient.Ping(ctx).Err()
```
this means the sync will terminate 5 seconds after it started